### PR TITLE
Feature/mac support

### PR DESCRIPTION
**Pull request by melekr**
- Adds native macOS support

ref: BT-1029

### CODE REVIEW

File: Runtime/Model/Attributes/MachineAttributeProvider.cs [hunk 1]
```diff
@@ -90,6 +90,10 @@ private void IncludeOsInformation(IDictionary<string, string> attributes)
             attributes["uname.version"] = UnityEngine.iOS.Device.systemVersion;
             attributes["uname.fullname"] = Environment.OSVersion.Version.ToString();
 #endif
+
```
Comment (konraddysput) [inline Runtime/Model/Attributes/MachineAttributeProvider.cs:93]: why we cannot use the if statement above ?

File: Runtime/Native/NativeClientFactory.cs [hunk 2]
```diff
@@ -22,8 +22,10 @@ internal static INativeClient CreateNativeClient(BacktraceConfiguration configur
             return new Android.NativeClient(configuration, breadcrumbs, attributes, attachments, gameObjectName);
 #elif UNITY_IOS
             return new iOS.NativeClient(configuration, breadcrumbs, attributes, attachments);
+#elif UNITY_STANDALONE_OSX
+            return new OSX.NativeClient(configuration, breadcrumbs, attributes, attachments);
 #else
-            return null;
+                return null;
```
Comment (konraddysput) [inline Runtime/Native/NativeClientFactory.cs:28]: additional tab

**Review by konraddysput — COMMENTED** [review]
Did you verify OOMs?

File: Runtime/Model/Attributes/MachineAttributeProvider.cs (hunk 1 above)
Comment (melekr) [reply Runtime/Model/Attributes/MachineAttributeProvider.cs:93]: UnityEngine.iOS.Device.systemVersion is iOS specific

**Comment by melekr** [comment]
> Did you verify OOMs?

OOM generates a crash report with signal code : SEGV_ACCERR and Error Message: Segmentation fault: 11.

File: Runtime/Model/Attributes/MachineAttributeProvider.cs (hunk 1 above)
Comment (konraddysput) [reply Runtime/Model/Attributes/MachineAttributeProvider.cs:93]: Yeah but in talking about environment.osversion

**Review by konraddysput — APPROVED** [review]

**Pull request merged**